tests/unit/amo/components/TestLandingPage.js

```diff
@@ -17,31 +13,43 @@ import {
   SEARCH_SORT_TOP_RATED,
 } from 'core/constants';
 import I18nProvider from 'core/i18n/Provider';
+import { ErrorHandler } from 'core/errorHandler';
 import { visibleAddonType } from 'core/utils';
 import { dispatchClientMetadata, fakeAddon } from 'tests/unit/amo/helpers';
 import { getFakeI18nInst } from 'tests/unit/helpers';
+import ErrorList from 'ui/components/ErrorList';
 
 
 describe('<LandingPage />', () => {
-  function render({ ...props }) {
-    const { store } = dispatchClientMetadata();
-    const fakeDispatch = sinon.stub();
+  function renderProps(props = {}) {
+    return {
+      dispatch: sinon.stub(),
+      errorHandler: new ErrorHandler({
+        id: 'some-handler',
+        dispatch: sinon.stub(),
+      }),
+      i18n: getFakeI18nInst(),
+      params: { visibleAddonType: visibleAddonType(ADDON_TYPE_EXTENSION) },
+      resultsLoaded: false,
+      ...props,
+    };
+  }
 
-    return findRenderedComponentWithType(renderIntoDocument(
+  function render(props = {}) {
+    return shallow(<LandingPageBase {...renderProps(props)} />, {
+      lifecycleExperimental: true,
```

Comment:
Hah, this explains why you could use `componentDidUpdate` and have them work with the tests. Neato. 👍